### PR TITLE
Addded two unit tests to dataframe_functions

### PR DESCRIPTION
**Pull request by ayahusseini**
Addded two unit tests to dataframe_functions to cover the add_year_month_day_col method and give full coverage of the file.

### CODE REVIEW

File: dashboard/conftest.py [hunk 1]
```diff
@@ -6,6 +6,18 @@
 import pandas as pd
 
 
+@pytest.fixture
```
Comment (indiah444) [inline dashboard/conftest.py:9]: Love the use of fixtures!

File: dashboard/conftest.py [hunk 2]
```diff
@@ -6,6 +6,18 @@
...
+def fake_date_published():
+    return {
+        "date_published": [
+            pd.Timestamp("2024-10-18"),  # A random date
+            pd.Timestamp("2020-02-29"),  # Leap year date
+            pd.Timestamp("2024-01-01"),  # Start of the year
+            pd.Timestamp("2024-12-31")   # End of the year
+        ]
```
Comment (indiah444) [inline dashboard/conftest.py:17]: Thank you for adding these comments, makes the dates easier to understand.

File: dashboard/test_dataframe_functions.py [hunk 3]
```diff
@@ -1,10 +1,34 @@
+# pylint: skip-file
 """Write unit tests for dataframe functions"""
 import pytest
 from unittest.mock import patch
 
 import pandas as pd
 
-from dataframe_functions import resample_dataframe, is_valid_time_interval
+from dataframe_functions import resample_dataframe, is_valid_time_interval, add_year_month_day_columns
```
Comment (megansam5) [inline dashboard/test_dataframe_functions.py:8]: Nice unit tests wooo!

**Review by megansam5 — APPROVED** [review]
I think this all looks great!

**Pull request merged**